.github/workflows/simulation_test.yml

```diff
@@ -200,6 +199,9 @@ jobs:
           export HDF5_INCLUDEDIR=/usr/include/hdf5/mpich
           export HDF5_LIBDIR=/usr/lib/x86_64-linux-gnu/hdf5/mpich
         fi
+        # Upgrade setuptools again, to avoid build error of `project.license`
+        # It should have been done in the previous step, no idea why
```

Comment:
Was it because there was a cache hit, so the change in the step wasn't executed?
Perhaps the previous commit probably worked, just the cache needed to be removed first?

Reply:
The caches for MacOS haven't been built yet in `main` because of the failure when pushing my PR into `main`. 
In fact, it is NEURON that requires an older setuptools previously which doesn't work for h5py. 
It is complicated when using caches. NEURON uses its cache, and venv has another cache. I wasn't able to see this failure at the end of my PR probably because of using NEURON cache which wouldn't revert the setuptools version at some point.